src/components/InstalledApp.vue

```diff
@@ -82,10 +85,31 @@ export default {
       if (this.torOnly && window.location.origin.indexOf(".onion") < 0) {
         event.preventDefault();
         alert(`${this.name} can only be used over Tor. Please access your Umbrel in a Tor browser on your remote access URL (Settings > Tor > Remote Access URL) to open this app.`);
+        return;
+      }
+      if (this.isUninstalling || this.isOffline) {
+        event.preventDefault();
+        return;
       }
       return;
     }
   },
+  created() {
+    const checkIfAppIsOffline = async () => {
+      try {
+        await window.fetch(this.url, { mode: "no-cors" });
+        this.isOffline = false;
+        window.clearInterval(this.pollIfAppIsOffline);
+      } catch (error) {
+        this.isOffline = true;
+      }
+    };
+    this.pollIfAppIsOffline = window.setInterval(checkIfAppIsOffline, 3000);
```

Comment:
This is going to fire every 3 seconds regardless of if there's all ready a request in-flight.

Could potentially lead to a race condition where a request succeeds, kills the interval, then fails.

If that's something we want to account for an async while loop would be better.

Reply:
Great point! Changed it in 44c62823ce7ba47bb764922fa5dc51fdee233d79 to make sure we don't send multiple requests if there's already a request in-flight. Also shortned the interval to 1s as we no longer have to worry about multiple pending requests.

Async while loop sounds good too, but this seemed more straightforward plus I wasn't sure how to cleanly stop the loop from executing once the user is no longer on the apps/app listing page.

Reply:
>Async while loop sounds good too, but this seemed more straightforward plus I wasn't sure how to cleanly stop the loop from executing once the user is no longer on the apps/app listing page.

I think an async loop would lead to simpler code.

Instead of 

```js
  created() {
    let requestInFlight = false;
    const checkIfAppIsOffline = async () => {
      if (requestInFlight) {
        return;
      }
      requestInFlight = true;
      try {
        await window.fetch(this.url, { mode: "no-cors" });
        this.isOffline = false;
        window.clearInterval(this.pollIfAppIsOffline);
      } catch (error) {
        this.isOffline = true;
      }
      requestInFlight = false;
    };
    this.pollIfAppIsOffline = window.setInterval(checkIfAppIsOffline, 1000);
    checkIfAppIsOffline();
  },
  beforeDestroy() {
    window.clearInterval(this.pollIfAppIsOffline);
  },
```

it would be

```js
  created() {
    while(!this.appCheckCompleted) {
      try {
        await window.fetch(this.url, { mode: "no-cors" });
        this.isOffline = false;
        this.appCheckCompleted = true;
      } catch (error) {
        this.isOffline = true;
      }
    },
  }
  beforeDestroy() {
    this.appCheckCompleted = true;
  },
```

Reply:
bruhhhhhh so sick ❤️
love it! no more `setInterval` hereon, fixed in da42f19 (added a 1s delay in the loop because the requests on local network are too quick and cause the browser to freeze)